array.rb

```diff
@@ -258,4 +213,24 @@ def fetch_values(*indexes, &block)
     indexes.map! { |i| fetch(i, &block) }
     indexes
   end
+
+  with_yjit do
+    if Array.instance_method(:each).source_location == nil # preserve monkey patches
```

Comment:
This technically isn't comprehensive enough of a check since the money patch could be a C method:

```
$ ruby -e 'its = Kernel.instance_method(:itself); \
      Array.define_method(:each, its);\
      p Array.instance_method(:each).source_location'
nil
```
Maybe check with rb_method_basic_definition_p()?

Reply:
Fixed in https://github.com/ruby/ruby/pull/11955/commits/218776d8aa1a651151c88b449036ef0d23b12387